pkg/observability/trace.go

```diff
@@ -39,6 +40,9 @@ func Start() error {
 		resource.WithProcessRuntimeName(),
 		resource.WithProcessRuntimeVersion(),
 		resource.WithFromEnv(),
+		resource.WithAttributes(
+			semconv.ServiceName(serviceName),
+		),
```

Comment:
`resource.WithFromEnv()` gets called here. It can set the service name from the `OTEL_SERVICE_NAME` environment variable. That is the way that I think we should set the service name.

https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/resource/env.go#L40

Reply:
I realize you mentioned this in the PR description